minbpe/base.py

```diff
@@ -6,20 +6,21 @@
 some concessions are made for simplicity.
 """
 import unicodedata
+from collections import Counter
 
 # -----------------------------------------------------------------------------
 # a few helper functions useful for both BasicTokenizer and RegexTokenizer
 
-def get_stats(ids, counts=None):
+def get_stats(ids, counts=None) -> Counter:
     """
     Given a list of integers, return a dictionary of counts of consecutive pairs
     Example: [1, 2, 3, 1, 2] -> {(1, 2): 2, (2, 3): 1, (3, 1): 1}
     Optionally allows to update an existing dictionary of counts
     """
-    counts = {} if counts is None else counts
-    for pair in zip(ids, ids[1:]): # iterate consecutive elements
-        counts[pair] = counts.get(pair, 0) + 1
-    return counts
+    c = Counter(zip(ids, ids[1:]))
+    if counts is not None:
+        c.update(counts)
+    return c
```

Comment:
This looks great, I was just working on this... :) One suggestion is to use `itertools.pairwise`:

```suggestion
from collections import Counter
from itertools import pairwise

 # -----------------------------------------------------------------------------
 # a few helper functions useful for both BasicTokenizer and RegexTokenizer

 def get_stats(ids, counts=None) -> Counter:
     """
     Given a list of integers, return a dictionary of counts of consecutive pairs
     Example: [1, 2, 3, 1, 2] -> {(1, 2): 2, (2, 3): 1, (3, 1): 1}
     Optionally allows to update an existing dictionary of counts
     """
     c = Counter(pairwise(ids))
     if counts is not None:
         c.update(counts)
     return c
```